### PR TITLE
Add selinux-policy for systemd-pstore service

### PR DESCRIPTION
**Pull request by deepak-rawat**
systemd-pstore is a service to archive contents of pstore.

Signed-off-by: Deepak Rawat <drawat.floss@gmail.com>

### CODE REVIEW

File: policy/modules/system/systemd.te [hunk 1]
```diff
@@ -1178,6 +1185,14 @@ allow systemd_tmpfiles_t systemd_journal_t:file relabel_file_perms;
 allow systemd_tmpfiles_t systemd_tmpfiles_conf_t:dir list_dir_perms;
 allow systemd_tmpfiles_t systemd_tmpfiles_conf_type:file read_file_perms;
 
+allow systemd_tmpfiles_t systemd_pstore_var_lib_t:dir { relabelfrom relabelto manage_dir_perms };
+allow systemd_tmpfiles_t systemd_pstore_var_lib_t:file manage_file_perms;
+
+# Allow systemd-tmpfiles to enable pstore kernel parameters over sysfs
+# /sys/module/printk/parameters/always_kmsg_dump
+# /sys/module/kernel/parameters/crash_kexec_post_notifiers
+allow systemd_tmpfiles_t sysfs_t:file write_file_perms;
```
Comment (pebenito) [inline policy/modules/system/systemd.te:1194]: This needs a new interface like `dev_read_sysfs()` but for writing files.

**Review by pebenito — CHANGES_REQUESTED** [review]

File: policy/modules/system/systemd.te (hunk 1 above)
Comment (deepak-rawat) [reply policy/modules/system/systemd.te]: Thanks for the suggestion. Fixed.

File: policy/modules/system/systemd.te [hunk 2]
```diff
@@ -1178,6 +1185,14 @@ allow systemd_tmpfiles_t systemd_journal_t:file relabel_file_perms;
 allow systemd_tmpfiles_t systemd_tmpfiles_conf_t:dir list_dir_perms;
 allow systemd_tmpfiles_t systemd_tmpfiles_conf_type:file read_file_perms;
 
+allow systemd_tmpfiles_t systemd_pstore_var_lib_t:dir { relabelfrom relabelto manage_dir_perms };
```
Comment (pebenito) [inline policy/modules/system/systemd.te:1188]: CI had an error, the `relabelfrom relabelto` should be replaced with `relabel_dir_perms`.

**Review by pebenito — CHANGES_REQUESTED** [review]

File: policy/modules/system/systemd.fc [hunk 3]
```diff
@@ -40,6 +40,7 @@
 /usr/lib/systemd/systemd-update-done	--	gen_context(system_u:object_r:systemd_update_done_exec_t,s0)
 /usr/lib/systemd/systemd-user-runtime-dir	--	gen_context(system_u:object_r:systemd_user_runtime_dir_exec_t,s0)
 /usr/lib/systemd/systemd-user-sessions	--	gen_context(system_u:object_r:systemd_sessions_exec_t,s0)
+/usr/lib/systemd/systemd-pstore		--	gen_context(system_u:object_r:systemd_pstore_exec_t,s0)
```
Comment (pebenito) [inline policy/modules/system/systemd.fc:43]: Need to move to line 38 to put this in alphabetical order.

File: policy/modules/system/systemd.fc [hunk 4]
```diff
@@ -60,6 +61,7 @@
 /var/lib/systemd/coredump(/.*)?	gen_context(system_u:object_r:systemd_coredump_var_lib_t,s0)
 /var/lib/systemd/linger(/.*)?	gen_context(system_u:object_r:systemd_logind_var_lib_t,s0)
 /var/lib/systemd/rfkill(/.*)?	gen_context(system_u:object_r:systemd_rfkill_var_lib_t,s0)
+/var/lib/systemd/pstore(/.*)?	gen_context(system_u:object_r:systemd_pstore_var_lib_t,s0)
```
Comment (pebenito) [inline policy/modules/system/systemd.fc:64]: Goes up one line.

File: policy/modules/system/systemd.te [hunk 5]
```diff
@@ -242,6 +242,13 @@ init_system_domain(systemd_user_runtime_dir_t, systemd_user_runtime_dir_exec_t)
 type systemd_user_tmpfs_t;
 userdom_user_tmpfs_file(systemd_user_tmpfs_t)
 
+type systemd_pstore_t;
+type systemd_pstore_exec_t;
+init_system_domain(systemd_pstore_t, systemd_pstore_exec_t)
+
+type systemd_pstore_var_lib_t;
+files_type(systemd_pstore_var_lib_t)
```
Comment (pebenito) [inline policy/modules/system/systemd.te:250]: Moves up to line 170.

File: policy/modules/system/systemd.te [hunk 6]
```diff
@@ -1178,6 +1185,14 @@ allow systemd_tmpfiles_t systemd_journal_t:file relabel_file_perms;
 allow systemd_tmpfiles_t systemd_tmpfiles_conf_t:dir list_dir_perms;
 allow systemd_tmpfiles_t systemd_tmpfiles_conf_type:file read_file_perms;
 
+allow systemd_tmpfiles_t systemd_pstore_var_lib_t:dir { manage_dir_perms relabel_dir_perms };
+allow systemd_tmpfiles_t systemd_pstore_var_lib_t:file manage_file_perms;
```
Comment (pebenito) [inline policy/modules/system/systemd.te:1189]: This should go up to line 1177.

File: policy/modules/system/systemd.te [hunk 7]
```diff
@@ -1178,6 +1185,14 @@ allow systemd_tmpfiles_t systemd_journal_t:file relabel_file_perms;
...
+
+# Allow systemd-tmpfiles to enable pstore kernel parameters over sysfs
+# /sys/module/printk/parameters/always_kmsg_dump
+# /sys/module/kernel/parameters/crash_kexec_post_notifiers
+dev_write_sysfs(systemd_tmpfiles_t)
```
Comment (pebenito) [inline policy/modules/system/systemd.te:1194]: This should go to the bottom of the other `dev_` lines below.

**Review by pebenito — CHANGES_REQUESTED** [review]
Lines look good, only trivial style issues to put the lines in the correct places. Then we can merge.

File: policy/modules/system/systemd.fc (hunk 3 above)
Comment (deepak-rawat) [reply policy/modules/system/systemd.fc]: Sorry missed that everything is arranged by alphabetically. Thanks.

**Review by pebenito — APPROVED** [review]

**Pull request merged**